test/js/sql/sql.test.ts

```
@@ -606,6 +606,66 @@ if (isDockerEnabled()) {
     }
   });
 
+  test("#17798: it should handle inserting values of type text[] using array of values", async () => {
+    try {
+      await sql`
+    CREATE TEMPORARY TABLE test (
+      id INTEGER PRIMARY KEY,
+      name TEXT,
+      arr TEXT[]
+    )
+  `;
+
+      // Test data
+      const item = {
+        id: 1,
+        name: "test",
+        arr: ["a", "b"],
+      };
+
+      // Insert
+      await sql`INSERT INTO test ${sql(item)}`;
+
+      // Retrieve and verify
+      const result = await sql`SELECT * FROM test WHERE id = ${item.id}`;
+      expect(result[0]).toEqual(item);
+    } finally {
+      await sql`DROP TABLE test`;
+    }
+  });
+
+  test("#17798: it should handle updating values of type text[] using array of values", async () => {
+    try {
+      await sql`
+    CREATE TEMPORARY TABLE test (
+      id INTEGER PRIMARY KEY,
+      name TEXT,
+      arr TEXT[]
+    )
+  `;
+
+      // Test data
+      const item = {
```

can you add a test with invalid types in the array? null, undefined, Infinity, nan, numbers, objects etc?

---

for objects/arrays, should it stringify the values or just error out ? (i think it should error out)
i think it should error out in all cases besides undefined, null and string

---

I would test the postgres.js behavior, but from top of my head (not sure) erroring

---

apparently it errors out only in the case of passing objects, but for the other values it just stringify them including arrays ([1,2] => "1,2")

---

let me know what you think of the current state !